### PR TITLE
Adding structured codelist

### PR DESCRIPTION
**Pull request by timgdavies**
https://github.com/open-contracting/ocds-extensions/issues/43

### CODE REVIEW

**Comment by jpmckinney** [comment]
Travis (easiest if you scroll all the way down – I need to configure py.test to not be so verbose!) indicates that `openCodelist` should be `false`, since `enum` is set.

**Comment by jpmckinney** [comment]
Another test failed because `codelists` needs to be added in `extension.json`

**Comment by timgdavies** [comment]
Thanks @jpmckinney - just fixing that now. Was trying to work out from https://github.com/open-contracting/standard-maintenance-scripts how I can run those tests locally. 

Any chance we could add details of that to the README or to http://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/extensions/ ?

**Comment by jpmckinney** [comment]
I'll add it to the readme. But after running `pip install -r requirements.txt`, you can change into an extension directory and then run `py.test -rs path/to/standard-maintenance-scripts/tests/test_json.py`. (Travis runs other tests, but that's the one that typically fails.)

File: release-schema.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
             "title":"Paid by",
             "description":"Is this a user charge (paid by businesses or citizens using the facilities provided by the contract), or a charge paid by the government?",
             "type":["string","null"],
-            "enum":["government","user"]
```
Comment (jpmckinney) [inline release-schema.json]: Just confirming that it was not the original intent to have a closed codelist?

**Comment by timgdavies** [comment]
Fab - thanks. Will add that to my notes.

I had one other issue which I'm fixing now.

(Thanks for the tests BTW: really useful indeed to give me a refresher on the format and catch these rookie mistakes :)

File: extension.json [hunk 2]
```diff
@@ -10,5 +10,8 @@
   ],
   "documentationUrl": {
     "en": "https://github.com/open-contracting/ocds_charges_extension"
-  }
+  },
+  "codelists":[
```
Comment (jpmckinney) [inline extension.json:14]: Can you ensure consistent whitespace style? (space after colon, two spaces per indentation level)

File: release-schema.json (hunk 1 above)
Comment (timgdavies) [reply release-schema.json]: A good question. I don't think there is a strong intent either way. 

The PPP framework just specifies government vs user changes - and I'm assuming this is the basis for certain analysis to be done.

As the only use-case for this is currently the PPP framework, I've switched back to closed, as it is I think easier to relax to an open codelist in future, than to go the other way.

File: extension.json (hunk 2 above)
Comment (timgdavies) [reply extension.json:14]: Sure. Is there a warning we can easily add to the maintenance scripts that might check our preferred indentation has been applied?

Comment (jpmckinney) [reply extension.json:14]: When I was originally testing all JSON files, there were too many errors, but I'll try scoping it to specific filenames, or extensions only.

Comment (jpmckinney) [reply extension.json:14]: If I do that, I'll have to fix some files in other extensions myself anyway, so this is ready to merge.

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**